promql/engine.go

```diff
@@ -769,7 +769,7 @@ func (enh *EvalNodeHelper) dropMetricName(l labels.Labels) labels.Labels {
 	}
 	h := l.Hash()
 	ret, ok := enh.dmn[h]
-	if ok {
+	if ok && labels.Equals(l, ret) {
```

Comment:
Won't this test generally be false, as the input and output will be different?

Reply:
indeed. I would also need a map[2]labels.